tools/autograd/derivatives.yaml

```diff
@@ -606,7 +606,7 @@
 
 - name: index_fill_.int_Tensor(Tensor(a!) self, int dim, Tensor index, Tensor value) -> Tensor(a!)
   self: grad.clone().index_fill_(dim, index, 0)
-  value: grad.index_select(dim, index).sum()
+  value: grad.index_select(dim, std::get<0>(at::_unique(index, /*sorted=*/false))).sum()
```

Comment:
What happens during the forward when the same index is passed multiple times? Is one taken arbitrarily?

Reply:
index_fill fills all the specified locations with a single value, so result is deterministic, but the gradient depends on how many locations in the result tensor were changed, and those locations shouldn't be double-counted.

Reply:
Ho value is a scalar Tensor here all the time :D 

If it was the same size as index, we would have to know which value was actually written in the final output. But I was pessimistic!